code/app/src/main/res/layout/activity_add_recipe.xml

```diff
@@ -0,0 +1,150 @@
+        />
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/title"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="16dp"
+        android:layout_marginTop="16dp"
+        android:layout_marginEnd="16dp"
+        android:hint="Title"
+        app:layout_constraintBottom_toBottomOf="@+id/recipeCategory"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/addPhoto">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:inputType="text" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/recipeCategory"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="16dp"
+        android:layout_marginTop="16dp"
+        android:layout_marginEnd="16dp"
+        android:hint="Recipe Category"
+        app:layout_constraintEnd_toEndOf="@id/servings"
+        app:layout_constraintEnd_toStartOf="@+id/servings"
+        app:layout_constraintHorizontal_bias="0.5"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/title">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/servings"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
+        android:layout_marginEnd="16dp"
+        android:hint="Servings"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.5"
+        app:layout_constraintStart_toEndOf="@+id/recipeCategory"
+        app:layout_constraintTop_toBottomOf="@+id/title">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/addIngredients"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:hint="Add Ingredients"
+        android:layout_marginTop="16dp"
+        android:layout_marginEnd="16dp"
+        android:layout_marginStart="16dp"
+        app:layout_constraintTop_toBottomOf="@+id/recipeCategory">
+
+        <com.google.android.material.textfield.TextInputEditText
```

Comment:
`TexInputEditText`

Reply:
Do the same for all

Reply:
Can you elaborate on why you would like me to do that?

Reply:
To make it concise and consistent with the rest of the code!

Reply:
There is no TextInputEditText element. It has to be com.google.android.material.textfield.TextInputEditText.

Reply:
why do you need a textinputedittext element
wont edittext work

Reply:
Using com.google.android.material.textfield.TextInputEditText class allows u to display a hint in the IME when in 'extract' mode and provides accessibility support for TextInputLayout.
Ref: https://developer.android.com/reference/com/google/android/material/textfield/TextInputEditText

Clearer explanation with an example in Landscape view is here: https://stackoverflow.com/questions/35775919/edittext-added-is-not-a-textinputedittext-please-switch-to-using-that-class-ins

Reply:
I think textinputededittext gives the box format as given in the storyboards

Reply:
I think editText is only a line, whereas TextInputEditText is what i think we've on the storyboards

Reply:
If thats the case, only instructions and comments need to be TextInputEditText. Everything else is a line

Reply:
That works, but i feel like it would look odd with different background colors